### PR TITLE
RavenDB-12203: Delete DeleteOnClose==true mmaped files in Linux (comp…

### PR DESCRIPTION
**Pull request by aviviadi**
…ression.buffer files, which are mot PosixTempMMapPager, they are PosixMMapPager {DeleteOnClose=true})

### CODE REVIEW

File: src/Voron/Platform/Posix/PosixAbstractPager.cs [hunk 1]
```diff
@@ -47,5 +49,24 @@ protected internal override unsafe void PrefetchRanges(Win32MemoryMapNativeMetho
...
+        
+        protected override void DisposeInternal()
+        {
+            if (_fd != -1)
+            {
+                // note that the orders of operations is important here, we first unlink the file
+                // we are supposed to be the only one using it, so Linux would be ready to delete it
+                // and hopefully when we close it, won't waste any time trying to sync the memory state
+                // to disk just to discard it
+                if (DeleteOnClose)
```
Comment (ayende) [inline src/Voron/Platform/Posix/PosixAbstractPager.cs:94]: Need `madvise` here

Comment (aviviadi) [reply src/Voron/Platform/Posix/PosixAbstractPager.cs:94]: madvise should be used in the unmapping events, and maybe we can just fadvise instead : fadvise(fd, DONTNEED) ?

Comment (aviviadi) [reply src/Voron/Platform/Posix/PosixAbstractPager.cs:94]: okey, irrelevant. fadvise is not good here. 
We will:
if (DeleteOnClose == true)
  madvise(addr, DONTNEED);
munmap(addr);

File: src/Voron/Platform/Posix/PosixAbstractPager.cs [hunk 2]
```diff
@@ -47,5 +54,53 @@ protected internal override unsafe void PrefetchRanges(Win32MemoryMapNativeMetho
...
+        {
+            // 32 bits should override this method and call AbstractPager::ReleaseAllocationInfo
+            base.ReleaseAllocationInfo(baseAddress, size);
+            var ptr = new IntPtr(baseAddress);
+
+            if (DeleteOnClose)
+            {
+                if (Syscall.madvise(ptr, new UIntPtr((ulong)size), MAdvFlags.MADV_DONTNEED) != 0)
+                {
+                    _log.Info($"Failed to madvise MDV_DONTNEED for {FileName?.FullPath}");
```
Comment (grisha-kotler) [inline src/Voron/Platform/Posix/PosixAbstractPager.cs]: missing IsInfoEnabled

**Pull request merged**